sys/sysmon/include/sysmon/ThreadStatus.hpp

```diff
@@ -0,0 +1,91 @@
+///
+/// Copyright (C) 2022 COGIP Robotics association <cogip35@gmail.com>
+///
+/// This file is subject to the terms and conditions of the GNU Lesser
+/// General Public License v2.1. See the file LICENSE in the top level
+/// directory for more details.
+///
+///
+///
+/// @defgroup    sys_sysmon System monitoring
+/// @ingroup     sys
+/// @brief       System monitoring module
+///              Module used to monitor memory (thread stacks, heap and overall).
+///
+/// @{
+/// @file
+/// @brief       Public API for sysmon module
+///
+/// @author      Gilles DOFFE <g.doffe@gmail.com>
+///
+
+// System includes
+#include <cstddef>
+#include <string>
+
+// Project includes
+#include "MemoryStatus.hpp"
+#include "PB_ThreadStatus.hpp"
+#ifdef MODULE_UARTPB
+#include "uartpb/UartProtobuf.hpp"
+#endif
+
+#pragma once
+
+#ifndef THREADSTATUS_NAME_MAX_LENGTH
+#  define THREADSTATUS_NAME_MAX_LENGTH 64
+#endif
+
+namespace cogip {
+
+namespace sysmon {
+
+class ThreadStatus: public MemoryStatus {
+    /// Protobuf message type. Shortcut for original template type.
+    using PB_Message = PB_ThreadStatus<THREADSTATUS_NAME_MAX_LENGTH>;
+
+    public:
+        /// Constructor
+        ThreadStatus();
+
+        /// Get thread name
+        inline std::string name() const { return name_; };
+        /// Get thread pid
+        inline uint32_t pid() const { return pid_; };
+        /// Set thread name
+        inline void set_name(const std::string &name) { name_ = name; };
+        /// Set thread pid
+        inline void set_pid(const uint32_t pid) { pid_ = pid; };
+
+        /// Return the Protobuf message.
+        const PB_Message &pb_message(void) const { return pb_message_; };
+        /// Update Protobuf message
+        void update_pb_message(void);
+
+    private:
+        /// Thread pid
+        uint32_t pid_;
+        /// Thread name
+        std::string name_;
```

Comment:
You should not use std::string but etl::string to avoid dynamic allocation.

Reply:
done